src/Symfony/Bundle/WebProfilerBundle/Tests/EventListener/WebDebugToolbarListenerTest.php

```diff
@@ -344,6 +344,20 @@ public function testCspIsKeptEnabledIfDumperWasNotUsed()
         $this->assertEquals("<html><head></head><body>\nWDT\n</body></html>", $response->getContent());
     }
 
+    public function testNullContentTypeWithNoDebugEnv()
+    {
+        $response = new Response('<html><head></head><body></body></html>');
+        $response->headers->set('Content-Type', null);
+        $response->headers->set('X-Debug-Token', 'xxxxxxxx');
+
+        $event = new ResponseEvent($this->createMock(Kernel::class), new Request(), HttpKernelInterface::MAIN_REQUEST, $response);
+
+        $listener = new WebDebugToolbarListener($this->getTwigMock(), false, WebDebugToolbarListener::ENABLED, null);
+        $listener->onKernelResponse($event);
+
+        $this->assertNull($response->headers->get('Content-Type'));
```

Comment:
```suggestion
        $this->assertSame('<html><head></head><body></body></html>', $response->getContent());
```
Better test than the toolbar is not injected?

Reply:
I think every unit test should be atomic. This one checks header behavior only

Сould make one another test for toolbar check if its really necessary

Something like this
```php
    public function testToolbarWithNullContentTypeWithNoDebugEnv()
    {
        $response = new Response('<html><head></head><body></body></html>');
        $response->headers->set('Content-Type', null);
        $response->headers->set('X-Debug-Token', 'xxxxxxxx');

        $event = new ResponseEvent($this->createMock(Kernel::class), new Request(), HttpKernelInterface::MAIN_REQUEST, $response);

        $listener = new WebDebugToolbarListener($this->getTwigMock(), false, WebDebugToolbarListener::ENABLED, null);
        $listener->onKernelResponse($event);

        $this->assertSame('<html><head></head><body></body></html>', $response->getContent());
    }
```

Reply:
The point is that `$this->assertNull($response->headers->get('Content-Type'));` can never fail since the test set it itself and the code has no intent of modifying it.
Anyway, the point of the test is to ensure there's no warning or errors, so maybe just replace the assertion with `$this->expectNotToPerformAssertions();`?